task-maker-format/src/ioi/sanity_checks/att.rs

```diff
@@ -400,3 +402,42 @@ impl SanityCheck<IOITask> for AttWithNoCheck {
         Ok(())
     }
 }
+
+/// Check that the template and grader in att compile together
+#[derive(Debug, Default)]
+pub struct AttTemplatesShouldCompile;
+
+impl SanityCheck<IOITask> for AttTemplatesShouldCompile {
+    fn name(&self) -> &'static str {
+        "AttTemplatesShouldCompile"
+    }
+
+    fn pre_hook(&mut self, task: &IOITask, eval: &mut EvaluationData) -> Result<(), Error> {
+        for grader in task.grader_map.all_paths() {
+            let ext = grader
+                .extension()
+                .ok_or_else(|| anyhow!("Grader has no extension"))?
+                .to_string_lossy();
+            let att_name = format!("att/{}.{}", task.name, ext);
```

Comment:
This is assuming the name of the template files. I've checked the `oii` private repo:

- 186 tasks use this format (`<taskname>.cpp`)
- 7 have `template_cpp.cpp`
- 5 have `template.cpp`

I've also checked the `ois` repo, and there all the 366 tasks are with the `<taskname>.cpp` format.

I guess this is fine.